enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,257 @@
+  identified and discussed here.
+- Micro version upgrades - users should be able to skip forward versions within a
+  minor release stream without being required to pass through intermediate
+  versions - i.e. `x.y.N->x.y.N+2` should work without requiring `x.y.N->x.y.N+1`
+  as an intermediate step.
+- Minor version upgrades - you only need to support `x.N->x.N+1` upgrade
+  steps. So, for example, it is acceptable to require a user running 4.3 to
+  upgrade to 4.5 with a `4.3->4.4` step followed by a `4.4->4.5` step.
+- While an upgrade is in progress, new component versions should
+  continue to operate correctly in concert with older component
+  versions (aka "version skew"). For example, if a node is down, and
+  an operator is rolling out a daemonset, the old and new daemonset
+  pods must continue to work correctly even while the cluster remains
+  in this partially upgraded state for some time.
+
+Downgrade expectations:
+- If an `N->N+1` upgrade fails mid-way through, or if the `N+1` cluster is
+  misbehaving, it should be possible for the user to rollback to `N`. It is
+  acceptable to require some documented manual steps in order to fully restore
+  the downgraded cluster to its previous state. Examples of acceptable steps
+  include:
+  - Deleting any CVO-managed resources added by the new version. The
+    CVO does not currently delete resources that no longer exist in
+    the target version.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+## Implementation History
+
+2020-04-23: Provisional enhancement draft
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+### Only remove Jenkins from the OCP Payload
+
+OCP 3.x initially had a feature that auto-provisioned Jenkins if a JenkinsPipeline build was started.
+This capability was removed in the later OCP 3 releases, but its legacy was one reason that Jenkins was included in the OCP payload.
+
+Today users must deploy Jenkins via the provided templates to take advantage of OpenShift's integrations.
+The Samples Operator is responsible for installing the imagestream and templates for Jenkins, just like other Red Hat products delivered in this fashion.
+Jenkins is the only imagestream installed via the Samples Operator whose images exist in the payload.
```

Comment:
Maybe this is a good way forward initially. Get rid of the image, align the usage with the rest of samples. In parallel proceed with deprecation warnings everywhere with suggestions to use jenkins operator instead. 
A followup, would be to entirely drop the support once we have a viable replacement and users get sufficient time to migrate off the built-ins.

Reply:
+1